common/models/user.js

```diff
@@ -428,6 +428,117 @@ module.exports = function(User) {
   };
 
   /**
+   * Default verification options to use when calling User.verify()
+   * This is the full list of possible params, with example values
+   *
+   * ```js
+   * var options = {
+   *      type: 'email',
+   *      mailer: anEmailModel,
+   *      to: 'test@email.com',
+   *      from: 'noreply@email.com'
+   *      subject: 'verification email subject',
+   *      text: 'verification email text',
+   *      headers: {'Mime-Version': '1.0'},
+   *      template: 'path/to/template.ejs',
+   *      redirect: '/',
+   *      verifyHref: '',
+   *      host: 'localhost'
+   *      protocol: 'http'
+   *      port: 3000,
+   *      restApiRoot= '/api',
+   *      tokenGenerator: function (user, cb) { cb("random-token"); }
+   *    };
+   *```
+   *
+   * Usually you should only require to modify a subset of these params
+   * See `User.verify()` and `User.prototype.verify()` doc for params reference
+   */
+
+  User.verifyOptions = {
+    type: 'email',
+  };
+
+  /**
+   * Verify a user's identity by sending them a confirmation email.
+   *
+   * ```js
+   *    var options = {
+   *      type: 'email',
+   *      to: user.email,
+   *      template: 'verify.ejs',
+   *      templateFn: function(options, cb) { cb(...) }
+   *      redirect: '/',
+   *      tokenGenerator: function (user, cb) { cb(...); }
+   *    };
+   *
+   *    user.verify(options, next);
+   * ```
+   *
+   * @options {Object} options
+   * @property {String} type Must be 'email'.
+   * @property {String} to Email address to which verification email is sent.
+   * @property {String} from Sender email addresss, for example
+   *   `'noreply@myapp.com'`.
+   * @property {String} subject Subject line text.
+   * @property {String} text Text of email.
+   * @property {String} template Name of template that displays verification
+   *  page, for example, `'verify.ejs'.
+   * @property {Function} templateFn A function generating the email HTML body
+   * from `verify()` options object and generated attributes like `options.verifyHref`.
+   * It must accept the option object and a callback function with `(err, html)`
+   * as parameters
+   * @property {String} redirect Page to which user will be redirected after
+   *  they verify their email, for example `'/'` for root URI.
+   * @property {Function} generateVerificationToken A function to be used to
+   *  generate the verification token. It must accept the user object and a
+   *  callback function. This function should NOT add the token to the user
+   *  object, instead simply execute the callback with the token! User saving
+   *  and email sending will be handled in the `verify()` method.
+   * @callback {Function} fn Callback function.
+   * @param {Error} err Error object.
+   * @param {Object} object Contains email, token, uid.
+   * @promise
+   */
+
+  User.verify = function(email, verifyOptions, options, cb) {
```

Comment:
I see two ways how to identify the user we want to verify: 1) by providing User's id 2) by providing User's email.

I would personally prefer to use `id`, I think it makes it more difficult to spam a specific user, because the spammer would have to know user's id. I may be wrong though.

What are the advantages of using email instead of id?